examples/misc/test/language_tour/async_test.dart

```diff
@@ -0,0 +1,54 @@
+// ignore_for_file: type_annotate_public_apis
+import 'dart:async';
+
+import 'package:test/test.dart';
+import 'package:dartlang_examples_util/print_matcher.dart' as m;
+
+void main() {
+  test('syncGenerator', () {
+    // #docregion sync-generator
+    Iterable naturalsTo(n) sync* {
+      int k = 0;
+      while (k < n) yield k++;
+    }
+    // #enddocregion sync-generator
+
+    main() {
+      print(naturalsTo(3));
+    }
+
+    expect(main, m.prints('(0, 1, 2)'));
```

Comment:
Good tests! Note that we don't need `main()` functions, so they can be dropped.
The generator functions can be tested directly, e.g.:

```dart
    expect(naturalsTo(3), [0, 1, 2]);
```